src/activity/exec.rs

```diff
@@ -0,0 +1,113 @@
+use std::path::{PathBuf};
+use std::net::{SocketAddr, IpAddr, TcpStream, UdpSocket};
+use std::process::{Command, Stdio};
+use std::fs::{self, File, OpenOptions};
+use std::io::{Result, Error, ErrorKind, Write};
+use std::iter;
+use chrono::{Utc};
+
+use super::ActivityContext;
+use super::log::{ActivityLog, ActionLogInfo, ProcessLogInfo, FileLogInfo, NetLogInfo};
+use super::spec::{FileAction, FileOp, FileType, IpProto};
+
+pub fn process_action(name: &str, args: &Vec<String>, ctx: &ActivityContext) -> Result<ActivityLog> {
+    let proc = Command::new(name)
+        .args(args)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()?;
+
+    let mut cmdvec = vec![name.to_string()];
+    cmdvec.extend(args.iter().cloned());
```

Comment:
See comment below about Vec::clone()

Reply:
In fact, it might be better to eliminate the need for these entirely by changing what's stored in the structs to begin with. It's definitely something I thought about revisiting.